reproman/interface/docker.shim

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/perl
+#emacs: -*- mode: shell-script; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: t -*-
+#ex: set sts=4 ts=4 sw=4 noet:
+#
+#
+# This script is a shim for the docker executable. The shim intercepts
+# calls to the docker executable, runs a image id discovery process, and
+# then proceeds with the original call to the original docker executable.
+#
+require 5.002;
+
```

Comment:
Should we use `strict` and `warnings`?  (It wouldn't point to anything major in the current code; just a few redefined variables.)